### ISSUE TITLE
Receiver unshare triggers complete unshare from the provider side

### ISSUE THREAD

**Issue by yasharpm** [pondersource/oc-opencloudmesh#46]
Steps to reproduce:
1. Prepare the development/test environment using dev-stock
2. Log in as Marie on oc2.docker
3. Create a user called `Anne`
4. Create a group called `test` and add both Marie and Anne to that group
5. Log in as Einstein on oc1.docker
6. Share a file with `test@oc2.docker`
7. Accept the share with both Marie and Anne on oc2.docker
8. Remove the share while logged in as Marie
9. Verify that Einstein sees the file as not shared anymore
10. Verify that Anne does not have the received share anymore (it appears at first but after user tries to open it, OC removes the share)

Expected behavior:
- Einstein must be still sharing the file with `test@oc2.docker`
- Anne should still have access to the shared file

Report belongs to [this](https://github.com/pondersource/oc-opencloudmesh/commit/046814cfa5adc85ef506641b7ee3c6b1c5f0b5a4) state of the code.

**Comment by yasharpm** [comment]
The code has been copied from the federatedfilesharing remote user share feature. Which makes sense to send a notification to the provider after the receiver removes the share. But now we don't want that behavior anymore.

**Comment by yasharpm** [comment]
The unshare sends a `DELETE` to the local server's webdav. Which ends up in `apps/oc-opencloudmesh/opencloudmesh/lib/Files_Sharing/External/Mount.php` calling `removeShare` on the `AbstractManager`.

Nowhere in this sequence a notification is sent to the provider. But a notification is obviously being sent at some point. I think I should look at the consumers of the `DeclineShare` event that the external manager produces.

**Comment by yasharpm** [comment]
The even is received by the `Application` class of the `federatedfilesharing` app. It calls `sendDeclineShare` on the default `Notifications` class.

**Comment by yasharpm** [comment]
The notification is received by our `OcmController` and the notification case `NOTIFICATION_TYPE_SHARE_DECLINED`. Which then calls `declineShare` on `AbstractFedShareManager` which always calls
``` php
$this->federatedShareProvider->removeShareFromTable($share);
```

**Comment by yasharpm** [comment]
I think it makes more sense to prevent triggering a notification message to the provider. Fortunately the `DeclineShare` event is only used on one place which solely sends to notification to the provider.

**Comment by yasharpm** [comment]
OK I made the changes and it is fixed. Strangely the tests are not failing. And I have no complaints about that!

**Comment by yasharpm** [comment]
Closing this issue also closes #45.
